### PR TITLE
updated function to pull data from pins

### PR DESCRIPTION
**Pull request by CraigDemke**
The new function is designed to be versatile and efficient, capable of replacing the existing function while providing compatibility across different environments. Specifically, it will:

- Server Compatibility: Operate seamlessly on both servers, ensuring consistent performance and reliability.
- Local Functionality: Function locally by utilizing the keyring for secure credential management.
- Multi-functional Design: Incorporate four distinct functionalities within a single function, streamlining processes and reducing redundancy.
- Exported Function: Only the final functionality will be exported in the package, simplifying the interface and usage for end-users.

### CODE REVIEW

**Review by JoyLindsay — APPROVED** [review]
This looks good/I approve.

File: NAMESPACE [hunk 1]
```diff
@@ -18,6 +16,7 @@ export(set_pins_api_key)
 export(set_sis_db_password)
 export(set_sis_db_username)
 export(uth_clean_student_id)
+export(uth_get_data_from_pins)
```
Comment (MattNickodemus) [inline NAMESPACE]: Can we change the name of this function uth_get_data_from_pin? We don't want to switch back and forth between plural and singular.

**Review by MattNickodemus — APPROVED** [review]

**Comment by CSPierce01** [comment]
Looks good!

**Review by CandiPierce — APPROVED** [review]
It looks good.

**Review by andreabringhurst — APPROVED** [review]
I agree with the changes and comments noted in Code Review.

**Review by mkhassan-isu — APPROVED** [review]

File: NAMESPACE (hunk 1 above)
Comment (CraigDemke) [reply NAMESPACE]: I change the name of the function in issue #3

**Pull request merged**